proto/errors.go

```diff
@@ -23,12 +23,12 @@ import (
 	"github.com/cockroachdb/cockroach/util/retry"
 )
 
-// ResponseWithError is a tuple of a Response and an error. It is used
-// to pass around a Response with its associated error where that
+// ResponseWithError is a tuple of a BatchResponse and an error. It is
+// used to return a BatchResponse with its associated error where that
 // entanglement is necessary (e.g. channels, methods that need to
 // return another error in addition to this one).
 type ResponseWithError struct {
-	Reply Response
+	Reply *BatchResponse
```

Comment:
I'll leave it like this for now because having a BatchResponse makes little sense if there's an error (when we're done with `req -> (resp, err)`).